astroquery/alma/tests/test_alma.py

```diff
@@ -14,100 +22,333 @@ def data_path(filename):
+        " WHERE pol_states='/XX/'"
+    assert _gen_sql({'polarisation_type': 'Dual'}) == common_select + \
+        " WHERE pol_states='/XX/YY/'"
+    assert _gen_sql({'polarisation_type': 'Full'}) == common_select + \
+        " WHERE pol_states='/XX/XY/YX/YY/'"
+    assert _gen_sql({'polarisation_type': ['Single', 'Dual']}) == \
+        common_select + " WHERE (pol_states='/XX/' OR pol_states='/XX/YY/')"
+    assert _gen_sql({'polarisation_type': 'Single, Dual'}) == \
+        common_select + " WHERE (pol_states='/XX/' OR pol_states='/XX/YY/')"
+
+
+def test_query():
+    # Tests the query and return values
+    tap_mock = Mock()
+    empty_result = Table.read(os.path.join(DATA_DIR, 'alma-empty.txt'),
+                              format='ascii')
+    mock_result = Mock()
+    mock_result.to_table.return_value = empty_result
+    tap_mock.search.return_value = mock_result
     alma = Alma()
-    monkeypatch.setattr(alma, '_get_dataarchive_url', _get_dataarchive_url)
-    # monkeypatch instructions from https://pytest.org/latest/monkeypatch.html
-    monkeypatch.setattr(alma, '_request', alma_request)
-    # set up local cache path to prevent remote query
-    alma.cache_location = DATA_DIR
+    alma._get_dataarchive_url = Mock()
+    alma._tap = tap_mock
+    result = alma.query_region('1 2', radius=1*u.deg)
+    assert len(result) == 0
+    assert 'Project code' in result.columns
+    tap_mock.search.assert_called_once_with(
+        "select * from ivoa.obscore WHERE "
+        "(INTERSECTS(CIRCLE('ICRS',1.0,2.0,1.0), s_region) = 1) "
+        "AND calib_level>1 AND data_rights='Public'", language='ADQL')
 
-    # the failure should occur here
-    result = alma.query_object('eta carinae')
+    # one row result
+    tap_mock = Mock()
+    onerow_result = Table.read(os.path.join(DATA_DIR, 'alma-onerow.txt'),
+                               format='ascii')
+    mock_result = Mock()
+    mock_result.to_table.return_value = onerow_result
+    tap_mock.search.return_value = mock_result
+    alma = Alma()
+    alma._tap = tap_mock
+    with patch('astroquery.alma.tapsql.coord.SkyCoord.from_name') as name_mock:
+        name_mock.return_value = SkyCoord(1, 2, unit='deg')
+        result = alma.query_object('M83', public=False,
+                                   band_list=[3])
+    assert len(result) == 1
 
-    assert len(result) == 15
-    assert '2011.0.00497.S' in result['Project code']
+    tap_mock.search.assert_called_once_with(
+        "select * from ivoa.obscore WHERE "
+        "(INTERSECTS(CIRCLE('ICRS',1.0,2.0,0.16666666666666666), s_region) = 1) "
+        "AND band_list LIKE '%3%' AND calib_level>1 AND data_rights='Proprietary'",
+        band_list=[3], language='ADQL')
+    # check ObsCore columns in result are the same with old ones
+    row = result[0]
+    for item in _OBSCORE_TO_ALMARESULT.items():
+        if item[0] == 't_min':
+            assert Time(row[item[0]], format='mjd').strftime('%d-%m-%Y') ==\
```

Comment:
my changes cause a failure here, but I suspect that's because the data are out of date - alma_onerow contains data that aren't returned from alma at present.